### DISCUSSION TITLE
Change active learning model  during screening

### DISCUSSION THREAD

**Discussion by boer0107-zz** [asreview/asreview#580] [Q&A] [answered]
Can I change the active learning model during screening? When I have (almost) finished screening, I would like to switch models to see if this results in relevant papers that weren't shown by the default model. Or is this not how these models work?

**Comment by SagevdBrand** [comment]
Unfortunately, it is not possible to switch models during the screening process. However, you could export your results so far and upload them again to a new project, using the decisions you have made so far as prior knowledge. Then in the setup of the model settings you could for example switch to a different model, or query strategy. 

Please note that the defaults for the active learning model, are the defaults because there is [research](https://doi.org/10.31219/osf.io/w6qbg) showing that they work best. If you would like to look for papers that are not predicted to be the most relevant, but maybe are on the boundary of relevancy, I would suggest to change the query strategy, see [here](https://asreview.readthedocs.io/en/latest/features/pre_screening.html?#select-model) for some more information.

EDIT: Currently, research is being done on switching models!

**Comment by BartJanBoverhof** [comment, marked as answer]
Hi @boer0107, thanks for your question.

There are currently two ways to go about this. 

## Option 1: Prior knowledge
As Sofie rightfully mentioned, A way to switch models is to simply screen an X amount of papers with one model, until the point you wish to switch to a different model. You could simply export your results, and use them as prior knowledge for a newly chosen model. This approach is not the most elegant solution, for which a different option is offered.  

## Option 2: Model switching plugin 
A [Model-Switching-plugin ](https://github.com/JTeijema/asreview-plugin-model-switcher/blob/main/asreviewcontrib/models/model_switcher.py) was developed by @JTeijema . With this plugin, two different models may be specified, and switching will automatically occur at a a certain `switchpoint`, for which the current default is set at 100 iterations. This implies that, after the first model screened 100 documents, a switch to the second model is made. The current default models as implemented in this  plugin are a `Support vector machine` with a `Deep Convolutional neural network` switch, and a `Support vector machine` with a `Shallow Neural network` switch. You are, however, free to specify both the `switchpoint` and the combination of models you desire. 

Please note that currently, this plugin can solely be utilised in ASReview's [Simulation mode](https://asreview.readthedocs.io/en/latest/lab/simulation.html), and is thus not *(yet)* offered in [Oracle mode](https://asreview.readthedocs.io/en/latest/lab/oracle.html).

**Reply by RaySit** [reply]
Hi @BartJanBoverhof I am attempting to use Option 1 to realize model switching. I am testing with a set of 17,991 records, in which I have labeled 71 records, 28 of which are relevant. 

<img width="1905" alt="Screenshot 2024-07-03 at 01 15 23" src="https://github.com/asreview/asreview/assets/77477538/68cdd5b6-9b7d-416e-88ab-c97e84101f9b">

 However, when I export the results as either RIS, CSV, or EXCEL, and import the file back to ASReview as a new project, the prior knowledge section shows that there are "6245 relevant and 11575 irrelevant records". Basically every record has already been labeled.

<img width="1920" alt="Screenshot 2024-07-03 at 01 17 06" src="https://github.com/asreview/asreview/assets/77477538/857bd787-39ee-48ab-9bf2-ad42ff263bcc">

I am struggling to troubleshoot the issue. Has this ever happened before?

**Reply by Rensvandeschoot** [reply]
Dear @RaySit, this is weird... can you share the project file with us? asreview@uu.nl
I am happy to take a look if I can find out what goes wrong

**Reply by RaySit** [reply]
Thank you @Rensvandeschoot. I've just sent the email.
